shared/js/background/events.es6.js

```diff
@@ -476,6 +477,24 @@ browser.webNavigation.onCommitted.addListener(details => {
     }
 })
 
+/**
+ * For each completed page load, update the extension's action icon
+ */
+browser.webNavigation.onCompleted.addListener(details => {
+    // only update the icon when the outermost frame is complete
+    if (details.parentFrameId !== -1) return
```

Comment:
@sammacbeth is this the best way to check for the 'outer most frame'? 

It seems to work well in chrome + ff, but I wondered if there was another mechanism?

Reply:
For `webNavigation`, that's the best way, yes.